ansys/mapdl/core/license.py

```diff
@@ -0,0 +1,349 @@
+
+
+LOCALHOST = '127.0.0.1'
+
+
+def try_license_file():
+    
+    try:
+        license_file_checker()
+    except LicenseServerConnectionError:
+        # expected error, so let it raise
+        raise 
+    except:
+        # Rest of the cases, let's run secondary methods.
+        # print("Error found trying to read license file.")
+        return False
+    else:
+        return True 
+
+
+def license_file_checker():
+    licdebug_file = os.path.join(get_licdebug_path(), get_licdebug_name())
+    file_iterator = get_licdebug_msg(licdebug_file)
+
+    time_to_be_checking = 10 #seconds
+    max_time = datetime.time(0, 0, time_to_be_checking) #hours, minute, seconds
+
+    while max_time < datetime.datetime.now().time():
+        msg = next(file_iterator)
+
+        if is_denied_msg(msg):
+            license_path = re.findall( "(?<=License path:)(.*)(?=;\n)", msg)[0]
+            license_port = license_path.split('@')[0]
+            license_hostname = license_path.split('@')[1]
+            raise LicenseServerConnectionError(
+                head_message=f"Error connecting to {license_port}:{license_hostname}",
+                error_message=msg,
+                tail_message=f"Error found in file {get_licdebug_name()}")
+        
+    else:
+        print("Time is out for license checking.")
+
+def get_licdebug_path():
+    if os.name == 'nt': 
+        folder = os.getenv('TEMP')
+    elif os.name == 'posix':
+        folder = os.getenv('HOME')
+    else:
+        raise OSError
+    
+    return os.path.join(folder, '.ansys')
+
+
+def get_licdebug_name():
+    # Licdebug name convention:
+    # - For version 22.1 and above: `licdebug.$hostname.$appname.$version.out`
+    # - For version 21.2 and below: `licdebug.$appname.$version.out`
+
+    from ansys.mapdl.core.launcher import _version_from_path, get_ansys_path 
+    
+    name = 'licdebug'
+    hostname = socket.gethostname()
+    appname = 'FEAT_ANSYS'  #TODO: We need to make sure this is the type of feature we need to checkout.
+    # This is the type of license my client requests (Windows 10, 2021R2)
+    version = _version_from_path(get_ansys_path(allow_input=False))
+    ending = 'out'
+
+    if version < 221:
+        parts = (name, appname, version, ending)
+    else:
+        parts = (name, hostname, appname, version, ending)
+    
+    return '.'.join(parts)
+
+
+def get_licdebug_msg(licdebug_file):
+    f = open(licdebug_file)
+    f.seek(0, 2)  # Going to the end of the file. 
+
+    buffer = []
+    while True:
+        line = f.readline()
+        if line:
+            if buffer == []: # not empty
+                buffer.append(line)
+
+            else:
+                if line.startswith('\t\t'):
+                    buffer.append(line)
+                else:
+                    msg = ''.join(buffer)
+                    buffer = [line]  # Flushing buffer
+                    yield  msg 
+        else:
+            time.sleep(0.01)
```

Comment:
```suggestion
    with open(licdebug_file) as f:
        f.seek(0, 2)  # Going to the end of the file. 

        buffer = []
        while True:
            line = f.readline()
            if line:
                if buffer == []: # not empty
                    buffer.append(line)

                else:
                    if line.startswith('\t\t'):
                        buffer.append(line)
                    else:
                        msg = ''.join(buffer)
                        buffer = [line]  # Flushing buffer
                        yield  msg 
            else:
                time.sleep(0.01)
```

It is best to use context managers when dealing with files, since it is easy to forget to close the file. See: https://book.pythontips.com/en/latest/context_managers.html

Reply:
Good suggestion. Thank you!